src/common/interfaces/IEvent.ts

```diff
@@ -91,4 +91,5 @@ export type FLZ_DESIGNER_EVENT_ACTION =
     | "get-designer-themes"
     | "navigate-to-documentation"
     | "open-generic-dialog"
-    | "custom-sections-action";
+    | "custom-sections-action"
+    | "generate-section-text";
```

Comment:
change to something like "generate-ai-action" to include all other future events (with a switch case)